packages/app/obojobo-document-engine/src/scripts/viewer/components/viewer-app.js

```diff
@@ -335,8 +339,13 @@ export default class ViewerApp extends React.Component {
 			}).then(res => {
 				this.leaveEvent = res.value
 			})
-		} else {
-			APIUtil.postEvent({
+
+			return
+		}
+
+		// returning
+		if (this.leaveEvent) {
+			ViewerAPI.postEvent({
```

Comment:
ViewerAPI is not defined here

Reply:
bah, I stashed it off the branch I was working on - it will be :( Too many spinning plates

Reply:
I this was finally resolved upstream as ViewerAPI was added in